modules/monitoring-creating-cluster-id-labels-for-metrics.adoc

```diff
@@ -0,0 +1,97 @@
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: PROCEDURE
+[id="creating-cluster-id-labels-for-metrics_{context}"]
+= Creating cluster ID labels for metrics
+
+You can configure cluster ID labels for metrics for default platform monitoring and for user workload monitoring.
+
+For default platform monitoring, you add cluster ID labels for metrics in the `write_relabel` settings for remote write storage in the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace. 
+
+For user workload monitoring, you edit the settings in the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace.
+
+.Prerequsites
+
+* You have installed the OpenShift CLI (`oc`).
+* You have configured remote write storage.
+* *If you are configuring default platform monitoring components:*
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
+* *If you are configuring components that monitor user-defined projects:*
+** You have access to the cluster as a user with the `cluster-admin` role or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` `ConfigMap` object.
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. In the `writeRelabelConfigs:` section under `data/config.yaml/prometheusK8s/remoteWrite`, add cluster ID relabel configuration values:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write-endpoint.example.com"
+        <endpoint_authentication_credentials>
+        writeRelabelConfigs: <1>
+          - <relabel_config> <2>
+----
+<1> Add a list of write relabel configurations for metrics that you want to send to the remote endpoint.
+<2> Substitute the label configuration for the metrics sent to the remote write endpoint.
++
+The following sample shows how to forward a single metric called `my_metric` with the cluster ID label `cluster_id_1` in default platform monitoring:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write-endpoint.example.com"
+        writeRelabelConfigs:
+        - sourceLabels: [__name__]
+          regex: 'my_metric'
+          action: keep
+        - sourceLabels:
+          - __tmp_openshift_cluster_id__ <1>
+          targetLabel: cluster_id <2>
+          action: replace <3>
```

Comment:
line 71 to 73 should be deleted also, though there are valid in prometheus CRD, but not here.
line 77 action should not be in configmap, the action will be added automatically to prometheus CRD. Detail please refer to my test case or Dev's auto case
https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-49176 user/password:readonly/polarion
https://github.com/openshift/cluster-monitoring-operator/pull/1578/files#diff-b388a528a3665a46f4dfa8eb4a8c02de116acb8f85e4611f12f11dc337d7f7b5 from line 887 to 1040

Reply:
I agree on `my_metric`. This seems confusing here. We have such a case in our e2e tests just to cover a corner case. The description adds to the confusion. In its current version two metrics are forwarded: `my_metric` and `cluster_id` but they have no relation to each other beyond being send to the remote endpoint.

On leaving out the action I'm conflicted. It is correct that `replace` is the default action in case no action is specified. I am however usually a fan of being explicit, especially in documentation. @bburt-rh maybe the doc team has a policy how default values are normally treated in config fragments like this? Maybe its worth keep the action but pointing out that its is optional since its the default value?